### DISCUSSION TITLE
Verbose vs Concise output

### DISCUSSION THREAD

**Discussion by feraloid** [OpenGammaProject/.github#2] [Ideas]
I've started fiddling with adding an ESP32 front end to my OpenGamma Detector (to give extra portable functionality, but especially Bluetooth and Wifi) and was thinking that for applications where the output is being interpreted by another machine, the human-friendly serial outputs are harder to code around. In such cases it would helpful to have a setting to turn "verbose" (human-friendly) messages off and instead send "concise" response codes more akin to an AT command set; error codes would just be like "ERROR: 300"; changes are acknowledged by something like "MODE: ENERGY"; HELP could still be available. This was a functionality I remember from the days of dialup modems...giving away my age here.

This has been a great project that I am really enjoying it and appreciate all the work that went into it!

**Comment by NuclearPhoenixx** [comment]
Hi there, interesting idea. You can actually fairly easily change the serial commands and they're respective serial return values by changing the functions defined in the following lines: https://github.com/OpenGammaProject/Open-Gamma-Detector/blob/1ef5fa6e72ed949a36c21b06535cb93d5c6049d5/software/ogd_pico/ogd_pico.ino#L1344...L1369

Anyways, the existing output is pretty easy to understand as well if you don't wanna do that. You know the serial commands and what they do, and if they're successful you get back a `[#]` at the start of the return line. If not, you get `[!]` (https://github.com/OpenGammaProject/Open-Gamma-Detector/blob/1ef5fa6e72ed949a36c21b06535cb93d5c6049d5/software/ogd_pico/Helper.cpp#L70). So you might as well scrape for these characters at the start of each line and know if it failed or not.

**Comment by feraloid** [comment]
Thanks for your suggestions! Since I've got the source code, I can access all the relevant output text, but figured that perhaps it's a thing to consider in the future...but admittedly, kind of a specialty adjustment.
